### ISSUE TITLE
\Xwraplemma et \Xwrapcontent ne provoquent pas la prise en compte de la langue secondaire par polyglossia pour les coupures de mots dans l'apparat

### ISSUE THREAD

**Issue by badumont** [maieul/ledmac#880]
# Version

2020/05/31 v2.33.0

# TeX Engine

Ce bug affecte xelatex, pas lualatex.

# Description

Avec xelatex et polyglossia, les commandes `\Xwraplemma` et `\Xwrapcontent` ne suffisent pas à provoquer les coupures de mots dans l'apparat si celui-ci utilise une langue qui n'est pas `mainlanguage`. Ainsi, dans l'exemple ci-dessous, προφητεύσουσι est correctement coupé dans le corps du texte, tandis qu'il ne l'est pas du tout dans l'apparat.

# Minimal Working Example

```latex
\documentclass[]{article}
\usepackage{polyglossia}
\setmainlanguage[variant=french, autospacing=false, autospaceguillemets=true, frenchfootnote=true]{french}
\setotherlanguage[variant=ancient]{greek}
\newfontfamily\greekfont{IFAO-Grec Unicode}
\usepackage[series={A},noend,noeledsec,noledgroup]{reledmac}
\Xarrangement[]{paragraph}
\Xwraplemma{\textgreek}
\Xwrapcontent{\textgreek}

\begin{document}

\begin{greek}

  \beginnumbering

  \pstart \edtext{Ὁ}{\lemma{ὁ}\Afootnote{om. S J}}\footnoteA{blabla} Θεὸς γὰρ ἀψευδής ἐστιν ὁ Υἱὸς τοῦ Θεοῦ. Εἰ δὲ \edtext{ἀναισχυντεῖς}{\lemma{ἀναισχυντεῖς}\Afootnote{ἀπαισχυντῆς Pa}}, πάλιν προφηστεύσουσι. Εἶπεν ὅτι· \emph{Ὁ νόμος καὶ οἱ προφῆται ἕως Ἰωάννου} τοῦ Βαπτιστοῦ \edtext{λατρεύσουσι}{\lemma{λατρεύσουσι}\Afootnote{λατρεύουσιν Pa, λατρεύσωσι S J}} καὶ \edtext{οὐ}{\lemma{οὐ}\Afootnote{om. xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx S J}} \edtext{προφητεύσουσι}{\lemma{προφητεύσουσι}\Afootnote{προφητεύουσιν Pa, προφητεύσωσι S J}}. \pend

  \endnumbering

\end{greek}

\end{document}
```

# Workaround

Inverser les langues dans `\setmainlanguage` et `\setotherlanguage`, ce qui implique d'inclure l'ensemble du document dans l'environnement de la langue dite secondaire pour que l'introduction et les titres soient traités comme souhaité (ici, passer `french` à `setotherlanguage` et inclure tout le document dans un environnement `french`).

**Comment by maieul** [comment]
En fait il ne devrait même pas y avoir besoin de passer par `\Xwraplemma` et `\wrapcontent` pour que la langue du texte soit prise en compte par les notes (cf. si on les utilise pas mais qu'on compile avec lualatex). Je regarde.

**Comment by maieul** [comment]
Hum, c'est assez compliqué. On ne voit pas le problème habituellement, car rare sont les gens qui alternent entre les langues au sein d'une page.

Mais le problème concerne un aspect profond de construction des notes paragraphées... et j'avoue n'avoir aucune piste.

J'en disucte avec des spécialistes des arcanes de TeX... pas évident.

**Comment by badumont** [comment]
Si le problème ne peut pas être résolu facilement, peut-être serait-il possible de le signaler dans la documentation en indiquant qu'il n'affecte pas LuaLaTeX ? Il me semble qu'il peut concerner toutes les personnes qui utilisent un apparat critique classique en latin en éditant un texte dans une autre langue, notamment dans les cas où les polices diffèrent.

**Comment by maieul** [comment]
oui, enfin l'apparat critique en latin utilise des mots courts et des abréviations qui ne sont pas soumises aux règles de césures.

J'attend des nouvelles de @davidcarlisle qui aura peut être des pistes supplémentaires (bien que j'ai des doutes) et sinon oui je mettrais dans le manuel.

**Comment by maieul** [comment]
Il semblerait qu'avec babel le souci ne soit plus là.

https://github.com/maieul/ledmac/commit/81524cdb12dfa6d0e71beb3df872f307118c795a

**Comment by maieul** [comment]
@reutenauer

aurait tu une idée de pourquoi on obtient un comportement différent entre babel et polyglossia ?
